src/main/main-window.ts

```diff
@@ -1,97 +1,115 @@
-            minHeight: 900,
-            paintWhenInitiallyHidden: true,
-            webPreferences: {
-                nodeIntegration: true,
-                contextIsolation: false,
-                nodeIntegrationInSubFrames: true,
-                nodeIntegrationInWorker: true,
-                webSecurity: false,
-                backgroundThrottling: false,
-            },
-        });
-
-        this.window.once("ready-to-show", () => this.show());
-
-        this.window.webContents.on("render-process-gone", (event, details) => {
-            console.error(details);
-        });
-
-        this.window.webContents.setWindowOpenHandler(({ url }) => {
-            shell.openExternal(url);
-            return { action: "deny" };
-        });
-
-        this.window.webContents.session.webRequest.onBeforeSendHeaders((details, callback) => {
-            callback({ requestHeaders: { Origin: "*", ...details.requestHeaders } });
-        });
-
-        this.window.webContents.session.webRequest.onHeadersReceived((details, callback) => {
-            const obj = { responseHeaders: { ...details.responseHeaders } };
-            if (!obj.responseHeaders["Access-Control-Allow-Origin"] && !obj.responseHeaders["access-control-allow-origin"]) {
-                obj.responseHeaders["Access-Control-Allow-Origin"] = ["*"];
-            }
-            callback(obj);
-        });
-
-        watch(
-            () => this.settings.model.displayIndex,
-            (displayIndex) => this.setDisplay(displayIndex)
-        );
-
-        watch(
-            () => this.settings.model.fullscreen,
-            (fullscreen) => {
-                this.window.setFullScreen(fullscreen);
-                this.window.maximize();
-            }
-        );
-
-        if (process.env.ELECTRON_RENDERER_URL) {
-            this.window.loadURL(process.env.ELECTRON_RENDERER_URL);
-            this.window.webContents.openDevTools();
-        } else {
-            this.window.loadFile(path.join(__dirname, "../renderer/index.html"));
-        }
+import { settingsService } from "./services/settings.service";
+import { logger } from "./utils/logger";
+import { replayContentAPI } from "@main/content/replays/replay-content";
+
+declare const MAIN_WINDOW_VITE_DEV_SERVER_URL: string;
+declare const MAIN_WINDOW_VITE_NAME: string;
+
+const log = logger("main-window");
+
+export function createWindow() {
+    const settings = settingsService.getSettings();
+    log.info("Creating main window with settings: ", settings);
+
+    const primaryDisplay = screen.getPrimaryDisplay();
+    const mainWindow = new BrowserWindow({
+        title: "Beyond All Reason",
+        fullscreen: settings.fullscreen,
+        frame: true,
+        show: true,
+        minWidth: 1440,
+        minHeight: 900,
+        paintWhenInitiallyHidden: true,
+        webPreferences: {
+            preload: path.join(__dirname, "../build/preload.js"),
+            zoomFactor: zoomFactor(primaryDisplay),
```

Comment:
This seem to mess up scale on my HiDPI display when electron runs under x11 (`ELECTRON_OZONE_PLATFORM_HINT=x11`), when running under native wayland `ELECTRON_OZONE_PLATFORM_HINT=wayland` the scale is ok. I have to press ctrl+0 to reset zoom level after each restart.

Why is touching zoom level needed?

Reply:
I tried to keep aspect ratio and the overall size of the UI elements consistent across resolutions and dpi. But that's not it.
I think the way to go is to stop using px as a css unit, and to use rem. Need some more investigating.

I'm going to remove this for now.

Reply:
I don't think that's expected, e.g. I never want to have exactly the same size of elements as measured by real world mm across devices. I want things to be smaller on my 13" laptop, and larger on my 27" screen.

When I want to scale something, there are OS settings for that purpose.

rem/em won't be useful as they are still just size relative to some other element on site sized with absolute sizes px/pt/in, and those are automatically scaled by browser based on OS scaling, so that px isn't really a real pixel.

Reply:
While that may be true for a desktop app, for a game lobby though ... I think we want to stay consistent across display sizes. 

But I'm not 100% set on this, I'm no design expert. I have an ultra widescreen though and it looks dumb in fullscreen in the current implementation.

Reply:
I realized maybe I misunderstood, which option do you want:
- Absolute size in real world units like `cm` consistent across devices?
- The relative size consistent, so e.g. the height of the "SINGLEPLAYER" button be always let's say 5% of total window height?

Reply:
I think we want the same content displayed on any screen size. If I take SC2 as a reference I don't think it's "responsive" like a classic webapp. Electron by default behaves like a responsive webapp, and it looks dumb on my ultrawide, lots and lots of empty space. A discussion need to happen with the designers anyway, let's maybe move this discussion on Discord

https://www.youtube.com/watch?v=byBIOsDDZiE